src/fklearn/training/classification.py

```diff
@@ -207,6 +207,123 @@ def p(new_df: pd.DataFrame, apply_shap: bool = False) -> pd.DataFrame:
+        Step size shrinkage used in update to prevents overfitting. After each boosting step,
+        we can directly get the weights of new features. and eta actually shrinks the
+        feature weights to make the boosting process more conservative.
+        See the eta hyper-parameter in:
+        https://catboost.ai/docs/concepts/python-reference_parameters-list.html
+
+    num_estimators : int
+        Int in the range (0, inf)
+        Number of boosted trees to fit.
+        See the n_estimators hyper-parameter in:
+        https://catboost.ai/docs/concepts/python-reference_parameters-list.html
+
+    extra_params : dict, optional
+        Dictionary in the format {"hyperparameter_name" : hyperparameter_value}.
+        Other parameters for the CatBoost model. See the list in:
+        https://catboost.ai/docs/concepts/python-reference_catboostregressor.html
+        If not passed, the default will be used.
+
+    prediction_column : str
+        The name of the column with the predictions from the model.
+        If a multiclass problem, additional prediction_column_i columns will be added for i in range(0,n_classes).
+
+    weight_column : str, optional
+        The name of the column with scores to weight the data.
+    """
+    from catboost import Pool, CatBoostClassifier
+    import catboost
+
+    weights = df[weight_column].values if weight_column else None
+    params = extra_params if extra_params else {}
+    params = assoc(params, "eta", learning_rate)
+    params = params if "objective" in params else assoc(params, "loss_function", 'Logloss')
+
+    dtrain = Pool(df[features].values, df[target].values, weight=weights, feature_names=list(map(str, features)))
+
+    cat_boost_classifier = CatBoostClassifier(iterations=num_estimators, **params)
+    cbr = cat_boost_classifier.fit(dtrain, verbose=0)
+
+    def p(new_df: pd.DataFrame, apply_shap: bool = False) -> pd.DataFrame:
+
+        dtest = Pool(new_df[features].values, feature_names=list(map(str, features)))
+
+        pred = cbr.predict_proba(dtest)[:, 1]
+
+        if params["loss_function"] == "MultiClass":
+            col_dict = {prediction_column + "_" + str(key): value
+                        for (key, value) in enumerate(pred.T)}
+            col_dict.update({prediction_column: pred.argmax(axis=1)})
+        else:
+            col_dict = {prediction_column: pred}
+
+        if apply_shap:
+            import shap
+            explainer = shap.TreeExplainer(cbr)
+            shap_values = list(explainer.shap_values(new_df[features]))
+            shap_expected_value = explainer.expected_value
+
+            shap_output = {"shap_values": shap_values,
+                           "shap_expected_value": np.repeat(shap_expected_value, len(shap_values))}
+
+            col_dict = merge(col_dict, shap_output)
+
+        return new_df.assign(**col_dict)
+
+    p.__doc__ = learner_pred_fn_docstring("CatBoostClassifier", shap=True)
+
+    log = {'CatBoostClassifier': {
```

Comment:
the dict log keys usually have the same name as the docstring `catboost_classification_learner`